shared/binlog.c

```diff
@@ -589,13 +595,20 @@ binlog * binlog_get_saved(binlog * node_binlog) {
 
 	/* free'd either here (binlog_destroy) or by the caller (node_binlog_read_saved) */
 	bl = malloc(sizeof(struct binlog));
+	if (bl == NULL) {
+		fclose(file);
+		return NULL;
+	}
+	memset(bl, 0, sizeof(struct binlog));
 
 	elements_read = fread(bl, sizeof(struct binlog), 1, file);
+	bl->path = strdup(node_binlog->file_metadata_path);
+	bl->file_save_path = strdup(node_binlog->file_save_path);
 	fclose(file);
 
 	/* Make sure we sucessfully read one binlog struct */
 	if (elements_read != 1) {
-		binlog_destroy(bl,BINLOG_UNLINK);
+		binlog_destroy(bl,BINLOG_DELALL);
```

Comment:
Do we not need to perform BINLOG_UNLINK here?

Reply:
Correct, I'll set the value of BINLOG_ALL to 6 so it covers both the UNLINK and delete of the save file.

Reply:
Checking back it seems we don't really need the new flag BINLOG_DELALL and can just use the BINLOG_UNLINK for the purpose deleting the related binlog files.